### PR TITLE
Add set_maven_property support

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #71

### CODE REVIEW

**Comment by geoand** [comment]
Can you please execute

`./for-all-boosters.sh -fb master-no-parent -m crud,tomcat-logging,configmap set_maven_property "foo.version" "bar"` 

and make sure that `sed` works correctly on Mac?

File: for-all-boosters.sh [hunk 1]
```diff
@@ -752,6 +752,26 @@ run_cmd() {
     fi
 }
 
+set_maven_property() {
+    local -r propertyName=${1}
+    local -r propertyValue=${2}
+
+    if ! grep --quiet '<properties>' pom.xml; then
+      # add an empty properties section right after the dependencies section
+      sed -i.bak -e "s/\/dependencies>/\/dependencies>\n\n  <properties>\n  <\/properties>/g" pom.xml
```
Comment (gytis) [inline for-all-boosters.sh]: Isn't it more common to have properties before dependencies?

Comment (geoand) [reply for-all-boosters.sh]: That's probably true (and most likely what we do in the boosters), but I need to see if I can achieve that with sed

**Comment by gytis** [comment]
I get `./for-all-boosters.sh: line 1067: pushd: spring-boot-istio-security-booster: No such file or directory`

File: for-all-boosters.sh (hunk 1 above)
Comment (geoand) [reply for-all-boosters.sh]: Fixed

**Comment by geoand** [comment]
@gytis That is very weird... Any output before that?

**Comment by gytis** [comment]
```
== BYPASSING CHECK FOR LOCAL CHANGES ==

== Will use 'master' branch(es) instead of the default 'master,redhat' ==

== Will use 'spring-boot-istio-security-booster' booster(s) ==

> spring-boot-istio-security-booster
./for-all-boosters.sh: line 1067: pushd: spring-boot-istio-security-booster: No such file or directory
```

**Comment by gytis** [comment]
Istio secured was the only one I had setup locally

**Comment by gytis** [comment]
Ah wait, let me check one thing

**Comment by gytis** [comment]
I forgot to add `-l` to show where the booster is. Here's how the properties look after execution:
```
<properties>
    <skip.fmp>true</skip.fmp>
    <foo.version>bar</properties>
```

**Comment by geoand** [comment]
Thanks @gytis, that's what I expected to see!

**Comment by metacosm** [comment]
I think `yq` can deal with xml, maybe use it instead of `sed` to perform the changes?

**Comment by geoand** [comment]
@metacosm I wasn't aware of that, let me check. If it can can deal with XML then I will use it

**Comment by metacosm** [comment]
@geoand I think the version we ended up using does, at least I seem to remember reading about it in the doc though I think it supports it via transcoding so I don't know how well it works.

**Comment by geoand** [comment]
It does indeed, but it seems like the support for adding a tag isn't really good (or at least I haven't been able to find it so far).
I'll keep at it for a while and if I can't get it to work, I'll just leave it as is

**Comment by metacosm** [comment]
+1

**Comment by geoand** [comment]
Support for adding/changing props does exist, but the output doesn't look good (namespaces get messed up among others).

I'll stick with sed for the time being

**Pull request merged**